packages/markdown-docx/src/OOXMLTransformer/index.js

```diff
@@ -58,171 +87,178 @@ class OoxmlTransformer {
+                constructedNode = this.construtCiceroMarkNodeJSON(updatedProperties);
+
+                if (commonLength === 0) {
+                    rootNode.nodes = [...rootNode.nodes, constructedNode];
+                    rootNodesLength++;
+                } else if (commonLength === 1) {
+                    rootNode.nodes[rootNodesLength - 1].nodes = [
+                        ...rootNode.nodes[rootNodesLength - 1].nodes,
+                        constructedNode,
+                    ];
+                }
             }
+            this.JSONXML = [];
+            this.nodes = [...this.nodes, rootNode];
         }
-        return {
-            isHeading: false,
-        };
     }
 
     /**
-     * Return the node after parsing each element
+     * Traverses for properties and value.
      *
-     * @param {object} element the element of the document node
-     * @returns {Array|object} the nodes
+     * @param {Array}  node            Node to be traversed
+     * @param {object} nodeInformation Information for the current node
      */
-    deserializeElement(element) {
-        switch (element.name) {
-        case 'w:p':
-            if (element.elements !== undefined) {
-                const headingElement = this.getHeading(element.elements[0].elements[0]);
-                const lineBreak = this.isLineBreak(element.elements[0].elements[0]);
-                if (headingElement.isHeading) {
-                    return {
-                        $class: `${NS_PREFIX_CommonMarkModel}Heading`,
-                        level: headingElement.level,
-                        nodes: [this.deserializeElements(element.elements)[0]],
-                    };
-                }
-                if (!lineBreak) {
-                    return {
-                        $class: `${NS_PREFIX_CommonMarkModel}Paragraph`,
-                        nodes: this.deserializeElements(element.elements)
-                    };
-                }
-            }
-            return;
-        case 'w:sdt': {
-            const name = this.getName(element.elements);
-            const value = this.getValue(element.elements);
-            const elementType = this.getElementType(element.elements);
-            if (name && value && elementType) {
-                return {
-                    $class: 'org.accordproject.ciceromark.Variable',
-                    value,
-                    name,
-                    elementType,
-                };
-            }
-            return this.deserializeElements(element.elements);
-        }
-        case 'w:sym':
-            return {
-                $class: `${NS_PREFIX_CommonMarkModel}Softbreak`,
-            };
-        case 'w:r':
-            if (element.elements[0].name === 'w:rPr') {
-                let emphasisedTextFound = element.elements[0].elements.some(
-                    subElement => {
-                        return subElement.name === 'w:i';
+    findAndTraverseRunTimeNodes(node, nodeInformation) {
```

Comment:
The method's purpose is to get the properties of a node. A better name would be `fetchFormattingProperties`.